nexus/src/nexus.rs

```diff
@@ -2249,6 +2259,49 @@ impl Nexus {
     pub async fn session_hard_delete(&self, token: String) -> DeleteResult {
         self.db_datastore.session_hard_delete(token).await
     }
+
+    pub async fn updates_refresh_metadata(&self) -> Result<(), Error> {
+        let rack = self.as_rack();
+        let trust_root = self
+            .tuf_trusted_root
+            .as_ref()
+            .ok_or_else(|| Error::InvalidRequest {
+                message: "updates system not configured".into(),
+            })?
+            .clone();
+
+        let artifacts = tokio::task::spawn_blocking(move || {
+            crate::updates::read_artifacts(&rack, &trust_root)
+        })
+        .await
+        // first, the JoinError:
+        .map_err(|e| Error::InternalError { internal_message: e.to_string() })?
+        // next, the boxed dyn Error:
+        .map_err(|e| Error::InternalError {
+            internal_message: e.to_string(),
+        })?;
+
+        // FIXME: if we hit an error in any of these database calls, the available artifact table
+        // will be out of sync with the current artifacts.json. can we do a transaction or
+        // something?
+
+        let mut current_version = None;
+        for artifact in artifacts {
+            current_version = Some(artifact.targets_role_version);
```

Comment:
As a sanity-check -- if we see multiple values of `artifact.targets_role_version`, they should all be the same, right? (trying to mentally verify that the order of `artifacts` couldn't change behavior)

Reply:
Yeah, these will all be the same.

This is kind of messy, it's a last minute "well crap I need this value" that we can clean up when moving the upsert and delete operations into the same transaction.